src/EconomicStrategy/EconomicStrategyManager.ts

```diff
@@ -98,14 +98,29 @@ export class EconomicStrategyManager {
       ? (await this.smartGasEstimation(txRequest)) || average
       : average;
 
-    const minGasPrice = txRequest.gasPrice;
-    const maxGasPrice = minGasPrice.times(this.maxSubsidyFactor);
-
-    return currentNetworkPrice.greaterThan(maxGasPrice)
-      ? maxGasPrice
-      : currentNetworkPrice.lessThan(minGasPrice)
-      ? minGasPrice
+    const minProfitabilityPrice = await this.calculateMinProfitabilityGasPrice(txRequest);
+    const minGasPrice = txRequest.gasPrice.greaterThan(currentNetworkPrice)
+      ? txRequest.gasPrice
       : currentNetworkPrice;
+    const executionGasPrice = minGasPrice.greaterThan(minProfitabilityPrice)
+      ? minGasPrice
+      : minProfitabilityPrice;
+
+    return executionGasPrice;
+  }
+
+  public async calculateMinProfitabilityGasPrice(
+    txRequest: ITransactionRequest
+  ): Promise<BigNumber> {
+    const { bounty } = txRequest;
+    const gasAmount = this.util.calculateGasAmount(txRequest);
+    const minProfitability = this.strategy.minProfitability;
+
+    const paymentModifier = (await txRequest.claimPaymentModifier()).dividedBy(100);
+    const reward = bounty.times(paymentModifier).minus(minProfitability);
+
+    const minProfitabilityGasPrice = reward.dividedBy(gasAmount);
```

Comment:
Instead of assigning to variable and returning it next line, please: `return reward.dividedBy(gasAmount);` immediately. The following suggestion is based on linting error (visible on Travis build).

```
ERROR: /home/travis/build/ethereum-alarm-clock/timenode-core/src/EconomicStrategy/EconomicStrategyManager.ts:105:31 - Immediately return this expression instead of assigning it to the temporary variable "executionGasPrice".
ERROR: /home/travis/build/ethereum-alarm-clock/timenode-core/src/EconomicStrategy/EconomicStrategyManager.ts:122:38 - Immediately return this expression instead of assigning it to the temporary variable "minProfitabilityGasPrice".
```